store.go

```diff
@@ -40,4 +40,7 @@ type Store interface {
 	// Audit Log
 	GetAuditEntries() ([]*AuditEntry, error)
 	AddAuditEntry(entry *AuditEntry) error
+
+	// Pre-migration
+	DoPreMigration() error
```

Comment:
I don't think that the Store interface should be concerned with migrations. Especially since it only affects the JSONStore

Reply:
The other way would be to move ``DoPreMigration()`` in ``migrations.go`` and have it passed the ``store`` the ``Migrate()`` method is passed. Don't see it a problem but that would require a check whether we are using a JSON or a Boltdb, that bugged me a bit

Reply:
I see your point, but I don't think that the regular operations of the store would ever need to "DoPreMigration"? it's purely linked to migrations, so i think it belongs in `migrations.go`